src/include_exclude.rs

```diff
@@ -0,0 +1,76 @@
+use crate::config::TemplateConfig;
+use ignore::gitignore::{Gitignore, GitignoreBuilder};
+use ignore::Match;
+use std::path::Path;
+
+pub struct Matcher {
+    template_config: Option<TemplateConfig>,
+    include_matcher: Option<Gitignore>,
+    exclude_matcher: Option<Gitignore>,
+}
+
+impl Matcher {
+    pub fn new(
+        template_config: Option<TemplateConfig>,
```

Comment:
Given that this function is one big `if let`/`else`, I would split it into two:

* `new` that only takes `project_dir`
* `new_with_template_config` that takes both a `project_dir` and a `TemplateConfig`

Reply:
and leave the option matching to the client instead? makes sense